mimix1050rt: ethernet initialization corrected

agansari opened the pull request:
Fixes board PHY initialization.
ENET_RST and ENET_INT pins are also used as USER_LED/JTAG_TDI and
INT1_COMBO/JTAG_TDO. Initialization needs to set this pins to power and
interrupt for the PHY Transciver.
Fixes #11586.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

pfalcon commented:
@agansari : Great work! Will test later today.

Minor issue in the commit message:

> Initialization needs to set this pins

Should be "these pins".

And well, need to do something about that completely unrelated test escaping 128K of code space. I didn't receive even comments on https://github.com/zephyrproject-rtos/zephyr/issues/11548, but apparently, that needs to be done sooner rather than later, please see if you can get to it sooner than me, as I didn't even get to needed stuff in me very intermittent reading of i.MX1052 TRM.

jukkar commented on boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst:
```diff
@@ -129,7 +129,9 @@ The MIMXRT1050 SoC has five pairs of pinmux/gpio controllers.
 +---------------+-----------------+---------------------------+
```
Typos in commit msg

s/etherent/ethernet/
s/corected/corrected/

pfalcon commented:
This patch actually makes it worse - the link guaranteedly stays down for me. I describe my testing in a series of comments in https://github.com/zephyrproject-rtos/zephyr/issues/11586#issuecomment-445432354 . 

I thus recommend not to merge it, because with it, the ethernet 100% doesn't work for some people, whereas without this patch, at least link can be established more or less deterministically and board->host TX packets work.

dbkinder reviewed (approved):
+1 for doc changes

pfalcon commented:
Now, just enabling CONFIG_NET_SHELL=y in dumb_http_server leads to dead link again.

pfalcon commented:
Size of dumb_http_server with CONFIG_NET_SHELL=n: 69712, CONFIG_NET_SHELL=y: 98012. So, I wouldn't blame code size on the issue, unless imxrt memory setup is really wrong.

pfalcon commented:
But now what I do is set `CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=n`. In other words, I have:

~~~
+CONFIG_NET_SHELL=y
+CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=n
~~~

And, bingo - Eth link is up.

@nordic-krch: I told you adding that option won't be wasted effort! cc: @jarz-nordic too.

@agansari: So, it looks like there's some conflict with UART [interrupts] and ETH [interrupts]. Can you please look into that?

pfalcon commented:
Enabling also CONFIG_NET_STATISTICS=y, I confirm that the board doesn't receive any packets. Trying to do ping from board, the host sees ARP request for 192.0.2.2, but the board doesn't see reply to it.

@agansari, So, we made a full iteration and back to square: RX doesn't work for me, like it didn't before. But at least Eth link establishment now works reliably, and we know how to reproduce a failure when it doesn't.

pfalcon commented:
And now knowing the cause of link-down cases, as far as I can tell, "k_busy_wait(USEC_PER_MSEC);" doesn't make any difference, so I'd suggest dropping this second commit here, let's vote this PR up as it goes in the right direction, and let's continue with investigation why RX interrupts still don't fire for some folks.

pfalcon reviewed (changes requested):
Let's remove busy-waiting (unless there're indications that it's really needed). Giving -1 so it wasn't merged by accident with that.

agansari committed:
mimix1050rt: ethernet initialization corrected

Fixes board PHY initialization.
ENET_RST and ENET_INT pins are also used as USER_LED/JTAG_TDI and
INT1_COMBO/JTAG_TDO. Initialization needs to set these pins to power and
interrupt for the PHY Transciver. PHY reset works without busy waiting.
Moved pin initialization back to PRE_KERNEL_1 to fix UART bug and busy
wait not required.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

pfalcon commented:
recheck

pfalcon commented:
This fails CI due to unrelated issue:
~~~
-- Loading /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/boards/arm/nucleo_f756zg/nucleo_f756zg.dts as base
-- Overlaying /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/dts/common/common.dts
nucleo_f756zg.dts_compiled: ERROR (duplicate_label): /flash@8000000: Duplicate label 'flash0' on /flash@8000000 and /soc/flash-controller@40023c00/flash@8000000
ERROR: Input tree has errors, aborting (use -f to force output)
CMake Error at /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/dts.cmake:130 (message):
  command failed with return code: 2
~~~

@agansari, Can you please rebase to see it that helps?

agansari commented:
@pfalcon rebase worked
